rayon-core/src/test.rs

```diff
@@ -30,3 +33,53 @@ fn worker_thread_index() {
     let index = pool.install(|| pool.current_thread_index().unwrap());
     assert!(index < 22);
 }
+
+#[test]
+fn start_callback_called() {
+    let n_called = Arc::new(AtomicUsize::new(0));
+
+    let clone = n_called.clone();
+    let conf = Configuration::new()
+        .set_num_threads(16)
+        .set_start_handler(Arc::new(move |index| {
+            clone.fetch_add(1, Ordering::SeqCst);
+        }));
+    {
+        let pool = ThreadPool::new(conf).unwrap();
+    }
+
+    // We must have started at least one thread.
+    assert!(n_called.load(Ordering::SeqCst) > 0);
```

Comment:
why must we have started at least one hread here? Perhaps it would be good to do something in the pool. For example:

```rust
pool.install(|| {
    .. // this is running in a worker thread, so we can certainly assert that the count is >0
});
```